conda/plugins/types.py

```diff
@@ -93,14 +98,57 @@ class CondaVirtualPackage(CondaPlugin):
     :param name: Virtual package name (e.g., ``my_custom_os``).
     :param version: Virtual package version (e.g., ``1.2.3``).
     :param build: Virtual package build string (e.g., ``x86_64``).
+    :param override_entity: Can be set to either to "version" or "build", the corresponding
+                            value will be overridden if the environment variable
+                            ``CONDA_OVERRIDE_<name>`` is set.
+    :param empty_override: Value to use for version or build if the override
+                           environment variable is set to an empty string. By default,
+                           this is ``NULL``.
+    :param version_validation: Optional version validation function to ensure that the override version follows a certain pattern.
+
+    Note:
+    - Both ``version`` and ``build`` parameters can be provided
+      as either a string, ``None``, or a deferred callable returning a string or ``None``.
+    - ``NULL`` is a sentinel value representing an explicitly empty override (``""``),
+      which is distinct from ``None`` (meaning "environment variable not set").
     """
 
     name: str
-    version: str | None
-    build: str | None
+    version: str | None | Callable[[], str | None]
+    build: str | None | Callable[[], str | None]
```

Comment:
Both of the callables may return `NULL`:

```suggestion
    version: str | None | Callable[[], str | None | _Null]
    build: str | None | Callable[[], str | None | _Null]
```